### ISSUE TITLE
WCF integration calls wrong ctor of ServiceHost base in case of InstanceContextMode.Single

### ISSUE THREAD

**Issue by TheBigRic** [simpleinjector/SimpleInjector#197]
see this [stackoverflow question](http://stackoverflow.com/q/35729614/3294832).

In case of a singleton mode configured WCF service [this constructor ](https://msdn.microsoft.com/en-us/library/ms585487(v=vs.110).aspx) 
``` c#
public ServiceHost(
	object singletonInstance,
	params Uri[] baseAddresses
)
```
should be called otherwise WCF will try to create the type itself and it therefore needs a default constructor.

Currently [`SimpleInjectorServiceHost`](https://github.com/simpleinjector/SimpleInjector/blob/master/src/SimpleInjector.Integration.Wcf/SimpleInjectorServiceHost.cs) calls the most obvious ctor where a type is supplied, probably because you would expect WCF the call back into the container for the first resolve, but that is not the case!

**Comment by dotnetjunkie** [comment]
What would be the correct solution to fix this?

**Comment by dotnetjunkie** [comment]
What should we do when the user marked his service with `[ServiceBehavior(InstanceContextMode = InstanceContextMode.Single)]`, while either the registered type in Simple Injector or its applied decorators are not registered as singleton?

**Comment by dotnetjunkie** [comment]
Fixed in v3.1.3.
